### ISSUE TITLE
 Problem with expected log P in QQ plots

### ISSUE THREAD

**Issue by soumickmj** [Cloufield/gwaslab#34]
Hi,
I created the sumstats object containing minimum log P value 2.00001 and maximum is 10.7109. I'm specifying skip=2 during the plot creation. I understand skip won't effect the QQ plot. 
![image](https://github.com/Cloufield/gwaslab/assets/20525305/a47c6b55-3858-4229-bf2e-d3c56e02caf0)
However, the expected log 10 P should also start from 2, not from 0, just like the observed one. 
How can I modify this?
Thanks :)

**Comment by Cloufield** [comment]
Hi,
Sorry that I haven't got your ideas yet. 
As you said, skip won't affect the data in QQ plot. 
I am wondering if you mean that there are no variants with P < 0.01 (-log10p>2) in the raw dataset in the first place. So your aim is to change the expected range of P from (0,1) to (0,0.01) manually? 
In this case, maybe the calculation for lambda GC will also be changed?
Could you let me know the aim of this? Thanks!

**Comment by soumickmj** [comment]
Yes, exactly. The streak should start also form 2. Currently, looking at the QQ plot, it feels like there is an inflation. Which isn't true. It's just we are dropping those insignificant variants. Thanks!

Inviato da Outlook per Android<https://aka.ms/AAb9ysg>
________________________________

**Comment by Cloufield** [comment]
I guess a better way might be using all variants to plot? To be honest I am not sure if the truncating is appropriate or not. (I haven't seen any qq plot using truncated p values before). Please inform me if you have any references.
But I will update an option for setting this in the new version and let you know when I finished (within this week).

**Comment by soumickmj** [comment]
Ya, I totally agree with you. It's surely better to use all the variants. But our problem is that I'm creating this by combining nearly 50 phenotypes. So, the memory requirement is astronomical! I already tried the other options you told me in the past, but then the only feasible one was to drop these insignificant variants. With the skip option for the manhatten plot, that's exactly what we would also like. So, fits perfectly for manhatten plots. But it's destroying the QQs.

Inviato da Outlook per Android<https://aka.ms/AAb9ysg>
________________________________

**Comment by Cloufield** [comment]
Thanks for the explanation. That would be reasonable. I will add this soon.

**Comment by soumickmj** [comment]
Thank you so much for your support 😊

Inviato da Outlook per Android<https://aka.ms/AAb9ysg>
________________________________

**Comment by Cloufield** [comment]
Just updated in v3.4.17.
You can add `expected_min_mlog10p=2`. 
Note that lambda gc in this case is calculated using the median of the truncated expected p values.

<img width="800" alt="image" src="https://github.com/Cloufield/gwaslab/assets/40289485/009cfc07-52ab-4e58-a2e5-fd15af7118d2">

**Comment by soumickmj** [comment]
Oh amazing!
Blazing fast!
Thank you so much :)

**Comment by soumickmj** [comment]
Hi,
It's working fine, except for the lamda value.
When I'm looking at the induvial phenotypes, the labda value is close to 1.
But, when I'm plotting together, I'm getting a huge lambda value, as can be seen here:

![image](https://github.com/Cloufield/gwaslab/assets/20525305/4bd2d3cd-d021-4b88-9efb-c70add704102)

Any clue what might be going wrong?
Thanks :)
